engine/src/main/java/org/hibernate/search/engine/mapper/mapping/building/impl/ConfiguredIndexSchemaNestingContext.java

```diff
@@ -85,6 +85,55 @@ public <T> Optional<T> addIndexedEmbeddedIfIncluded(
 			String prefixToParse = unconsumedPrefix + definition.relativePrefix();
 			int afterPreviousDotIndex = 0;
 			int nextDotIndex = prefixToParse.indexOf( '.', afterPreviousDotIndex );
+			if ( !definition.relativePrefix().endsWith( "." )
+					&& filter.isExplicitlyExcluded( definition.relativePrefix() ) ) {
+				// NOTE: isExplicitlyExcluded has a side effect of marking the path as encountered in the filter
+				// if this path is determined to be excluded.
+				//
+				// Any cases where unconsumed prefix is empty or not will not affect the filtering since we are going to
+				// use a relative path to the filter anyway (see how we are building objectNameRelativeToFilter).
+				//
+				// If a path we are about to add has any dots anywhere in it, and at the end in particular --
+				// we will get into the while loop, and it will handle the filtering for us.
+				//
+				// BUT if the path that we are about to add doesn't end with a dot -- we will end up ignoring
+				// the full relative path; and we'll not check it. We need to make sure that none of the exclude paths
+				// in any ancestors will not lead to our definition-about-to be added, e.g.
+				//
+				// @Indexed
+				// class A {
+				//     @IndexedEmbedded(excludePaths="obj.prefix_")
+				//     B b;
+				// }
+				// class B {
+				//     @IndexedEmbedded(prefix="obj.prefix_")
+				//     C c;
+				// }
+				// class C { ... }
+				//
+				// When we'll be adding node `C c` checking of `obj.` will happen in the while loop, but not `obj.prefix_`
+				// and that is why we are checking it with this additional explicit exclude check.
+				//
+				// We cannot simply check if `obj.prefix_` is just included. If we had a model with includePaths like:
+				//
+				// @Indexed
+				// class A {
+				//     @IndexedEmbedded(includePaths="obj.prefix_someProperty")
+				//     B b;
+				// }
+				// class B {
+				//     @IndexedEmbedded(prefix="obj.prefix_")
+				//     C c;
+				// }
+				// class C { ... }
+				//
+				// when we'd be adding `C c` we will end up checking `obj.prefix_` if it is included. BUT because we have
+				// an include path -- this sub-path will result in a "false-positive" exclude from `filter.isPathIncluded()`
```

Comment:
discussed here https://hibernate.zulipchat.com/#narrow/stream/132092-hibernate-search-dev/topic/HSEARCH-1182.20IndexEmbedded.20exclude.20paths/near/367837397
(incorrect expectations in a prefix-no-dots case, hence this will go away)